softwareComponents/simplesimLib/include/simplesim/module_states.hpp

```diff
@@ -261,10 +263,11 @@ class ModuleStates {
                           assert( configPtr );
                           return initInnerStatesFromConfiguration( *configPtr, verbose );
                       } ) )
+            , _collModel( collModel )
     {
-        assert( _physicalModulesConfiguration.visit( []( const auto & configuration ) {
+        assert( _physicalModulesConfiguration.visit( [ collModel ]( const auto & configuration ) {
```

Comment:
I feel like taking by reference here would be better, but since it's a shared_ptr, it's probably just a personal preference.